src/ckeditorcontext.jsx

```diff
@@ -13,34 +13,59 @@ export default class CKEditorContext extends React.Component {
 	constructor( props, context ) {
 		super( props, context );
 
+		/**
+		 * @type {module:watchdog/contextwatchdog~ContextWatchdog|null}
+		 */
 		this.contextWatchdog = null;
 
 		if ( this.props.isLayoutReady ) {
-			this._initializeContextWatchdog( this.props.config );
+			this._initializeContextWatchdog();
 		}
 	}
 
+	/**
+	 * Checks whether the component should be updated by React itself.
+	 *
+	 * @param {Object} nextProps
+	 * @return {Boolean}
+	 */
 	shouldComponentUpdate( nextProps ) {
-		// If the configuration changes then the ContextWatchdog needs to be destroyed and recreated
-		// On top of the new configuration.
+		// If the config has changed then the `ContextWatchdog` needs to be destroyed and recreated using the new config.
 		if ( nextProps.id !== this.props.id ) {
-			if ( this.contextWatchdog ) {
-				this.contextWatchdog.destroy();
-			}
-
-			this._initializeContextWatchdog( nextProps.config );
+			return true;
 		}
 
+		// Rerender the component once again if the layout is ready.
 		if ( nextProps.isLayoutReady && !this.contextWatchdog ) {
-			this._initializeContextWatchdog( nextProps.config );
-
 			return true;
 		}
 
 		// Rerender the component only when children has changed.
 		return this.props.children !== nextProps.children;
 	}
 
+	/**
+	 * Re-render the entire component once again. The old editor will be destroyed and the new one will be created.
+	 */
+	componentDidUpdate() {
+		// If the `isLayoutReady` property has changed from `false` to `true`, the instance of `ContextWatchdog` does not exist.
+		if ( this.contextWatchdog ) {
+			this.contextWatchdog.destroy();
+		}
+
+		this._initializeContextWatchdog( this.props.config );
+	}
+
+	/**
+	 * Destroy the context before unmounting the component.
+	 */
+	componentWillUnmount() {
+		this._destroyContext();
+	}
+
+	/**
+	 * @return {JSX.Element}
```

Comment:
TBH it's the React Component API. We don't need to document these methods :sweat\_smile: